redpanda/templates/_helpers.tpl

```diff
@@ -78,15 +82,15 @@ Generate configuration needed for rpk
 {{- end -}}
 
 {{- define "redpanda.kafka.internal.advertise.port" -}}
-{{- (first .Values.config.redpanda.kafka_api).port -}}
+{{- (first .Values.listeners.kafka.endpoints).port -}}
 {{- end -}}
 
 {{- define "redpanda.kafka.external.listen.address" -}}
 {{- "$(POD_IP)" -}}
```

Comment:
This could be a problem if this template is used in a file other than a pod that has $POD_IP defined.